Test failure in 3.19.0

I got the following test failure since the upgrade to 3.19.0:

```
======================================================================
FAIL: test__insert_do_not_modify_input (tests.test__collection_api.CollectionAPITest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-mongomock/src/mongomock-3.19.0/tests/test__collection_api.py", line 4743, in test__insert_do_not_modify_input
    self.assertEqual(
AssertionError: 946722630012 != 946719030012.0

----------------------------------------------------------------------
Ran 679 tests in 0.395s
```

Interestingly the value differs on different machine so it could be a timezone issue? The one above is a UTC+1 machine, and the one below is on a UTC+8 one:

```
======================================================================
FAIL: test__insert_do_not_modify_input (tests.test__collection_api.CollectionAPITest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-mongomock/src/mongomock-3.19.0/tests/test__collection_api.py", line 4743, in test__insert_do_not_modify_input
    self.assertEqual(
AssertionError: 946722630012 != 946693830012.0

----------------------------------------------------------------------
Ran 679 tests in 0.346s
```

--- a/tests/test__collection_api.py
+++ b/tests/test__collection_api.py
@@ -2,7 +2,6 @@ import collections
 import copy
 from datetime import datetime, tzinfo, timedelta
 from distutils import version  # pylint: disable=no-name-in-module
-import math
 import platform
 import random
 import re
@@ -5156,8 +5155,10 @@ class CollectionAPITest(TestCase):
             self.assertNotEqual(
                 original_document['date'].timestamp(), stored_document['date'].timestamp())
             self.assertEqual(
-                math.floor(original_document['date'].timestamp() * 1000),
-                stored_document['date'].timestamp() * 1000)
+                datetime(2000, 1, 1, 10, 30, 30, 12000),
+                stored_document['date'],
+                msg='The stored document holds a date as timezone naive UTC and without '
+                'microseconds')
 
         # The objects are not linked: modifying the inserted document or the fetched one will
         # have no effect on future retrievals.

